Server/src/models/DatasetModels/DatasetApprovalModel.ts

```diff
@@ -0,0 +1,88 @@
+    constructor() {
+        this.connection = getConnection();
+        this.commonModel = new DatasetCommonModel();
+    }
+
+    private selectDatasetCommentQuery = (id: number) =>
+        this.connection.createQueryBuilder(Dataset, 'dataset')
+            .select('dataset.comments', 'comments')
+            .where('dataset.id = :id', { id: id })
+            .getRawOne();
+
+    async selectUserFlaggedDatasets(uploaderId: number): Promise<IDatasetIDModel[]> {
+        let userFlaggedDatasets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_Datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .innerJoin(Dataset, 'dataset.id = unapproved_Datasets.datasetId')
+            .where('unapproved_datasets.isFlagged = 1')
+            .andWhere('dataset.uploaderId = :uploaderId', { uploaderId: uploaderId })
+            .getRawMany();
+        return userFlaggedDatasets
+    }
+
+    async getUnapprovedDatasets(): Promise<IDatasetIDModel[]> {
+        let allUnapprovedDatasets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .getRawMany();
+        return allUnapprovedDatasets
+    }
+
+    async selectAllFlaggedDatasets(): Promise<IDatasetIDModel[]> {
+        let allFlaggedDatsets = await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .select('unapproved_datasets.datasetId', 'dataset_id')
+            .where('unapproved_datasets.isFlagged = 1')
+            .getRawMany();
+        return allFlaggedDatsets
+    }
+
+    async updateApprovedStatus(datasetId: number) {
+        await this.connection.createQueryBuilder(Dataset, 'dataset')
+            .update('dataset')
+            .set({ isApproved: 1 })
+            .where('dataset.id = :datasetId', { datasetId: datasetId })
+            .execute()
+    }
+
+    async flagDataSet(datasetId: number, flaggedComment?: string, additionalComment?: string) {
+        await this.updateDatasetComments(datasetId, additionalComment)
+        await this.connection.createQueryBuilder(Unapproveddatasets, 'unapproved_datasets')
+            .update('unapproved_datasets')
+            .set({ flaggedComments: flaggedComment, isFlagged: 1 })
+            .where('unapproved_datasets.datasetId = :datasetId', { datasetId: datasetId })
+            .execute()
+        return "Dataset Flagged!"
+    }
+
+    async approveDataset(datasetId: number) {
+        await this.updateApprovedStatus(datasetId)
+        await this.commonModel.wipeEntryFromUnapprovedTable(datasetId)
+        return "Successfully approved new Dataset"
+    }
+
+    async updateDatasetComments(datasetId: number, datasetCommentsToAppend?: string) {
+        let oldComment = await this.selectDatasetCommentQuery(datasetId)
+        let newComment = oldComment.comments.concat(" " + datasetCommentsToAppend)

```

Comment:
so comments are getting appended? shouldnt we just override the previous comment? why is it appended

Reply:
did you test that it works the way you want it to? because concat accepts array as an argument https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat
and a string is an array of chars, if you wanna append one to an array use .push(string) instead

Reply:
The user and kiyandas comments are both relevant, kiyanda just wants the ability to add his own comment on top of the dataset, not override the users

Reply:
It is supposed to append as per Kiyanda's requirement